Fix Shared RP Documentation

razo7 opened the pull request:
### Which issue this PR addresses:

<!--
Please include a link to the ADO work item as well as any GitHub issues.

Usage: `Fixes #<GitHub issue number>`, or `Fixes (paste link of issue)`.
-->
It fixes the shared RP documentation.

### What this PR does / why we need it:

<!--
Include a brief summary of what the PR is intended to accomplish and how the PR
does it. (2-3 sentences)
-->

- Fix small caveats for creating the `env` file (missing export, and consistent string declaration)
- Fix indentation levels, spaces, and numbering.
- Reorganize the documentation file by moving the non-required procedures to the end under a new section.

### Test plan for issue:
Not required but the documentation will be used in a new planned automation
<!--
How did you test that this PR works?

- Are there unit tests?
- Are there integration/e2e tests?
- If it is not possible to write automated tests, explain why and document how
  to manually test and verify the feature.
-->

### Is there any documentation that needs to be updated for this PR?
Yes
<!--
- If yes and the docs are in GitHub, include doc updates in the PR.
- If yes and the docs are not in GitHub (i.e. ADO wiki), include a link to the
  docs.
- If no, explain why (e.g. "tech debt cleanup, N/A").
-->

### How do you know this will function as expected in production? 

<!--
- Does adequate telemetry, monitoring and documentation exist to effectively operate your change?
- Have failure modes been identified and mitigated? 
-->

razo7 commented on docs/prepare-a-shared-rp-development-environment.md:
```diff
@@ -418,20 +367,19 @@ az ad app credential reset \
+   export PROXY_HOSTNAME='vm0.$PROXY_DOMAIN_NAME_LABEL.\$LOCATION.cloudapp.azure.com'
+   export DATABASE_NAME='\$AZURE_PREFIX'
    export RP_MODE='development'
    export PULL_SECRET='$PULL_SECRET'
```
I am pretty confident that USER_PULL_SECRET is missing here, but IDK how to create/find it.

hlipsig replied:
Does make secrets get you the pull secret?

SrinivasAtmakuri replied:
The pull secret mentioned here only consists for tokens for prod and int svc acr and not to say other registries like redhat or quay, these can provided to cluster during install or at later stages, RP doesn't need them and not a blocker for cluster creation.

razo7 replied:
>Does make secrets get you the pull secret?

It does, but I am concerned about the **USER_PULL_SECRET** which is missing in the docs and exists in practice when you run `SECRET_SA_ACCOUNT_NAME=rharosecretsdev make secrets`.

razo7 commented:
/azp run ci,

hlipsig reviewed (commented):
I think these changes do look correct. I'll ask for one more person to confirm. Please see my comment on Pull Secret and let me know if you get that solved.

razo7 commented:
/azp run e2e

razo7 commented:
/azp run ci,e2e

razo7 commented:
/azp run e2e

razo7 commented:
I have updated the PR according to your great tip from Slack on using single quotes.
>what’s confusing me is the single quotes on the exported variable which includes escaped $ . From my understanding this basically means you creating a literal string $RESOURCEGROUP_PREFIX…. Am I missing something here?

I have addressed that in the last commit. Let me know if that's ok and if can we merge it now.
CC @jaitaiwan

jaitaiwan commented on docs/prepare-a-shared-rp-development-environment.md:
````diff
@@ -396,42 +344,51 @@ az ad app credential reset \
    ```bash
    # use a unique prefix for Azure resources when it is set, otherwise use your user's name
    cat >secrets/env <<EOF
+   ######## Prior of sourcing the file the following env vars must be set:    ########
````
I think this should say "Prior to sourcing"

jaitaiwan reviewed (approved):
Looks good to me. One small spelling error.

razo7 commented:
/azp run ci,e2e
